common/math_test.go

```diff
@@ -0,0 +1,77 @@
+package common
+
+import (
+	"golang.org/x/exp/constraints"
+	"golang.org/x/exp/slices"
+	"testing"
+)
+
+func intContains(s []int, v int) bool {
+	for i := range s {
+		if s[i] == v {
+			return true
+		}
+	}
+	return false
+}
+
+func genericContains[E comparable](s []E, v E) bool {
```

Comment:
If we can go over what E is that would be helpful for me!

Reply:
A Tour of Go's new section of generics has been helpful: https://go.dev/tour/generics/1

Reply:
Yes! There is also a more in-depth tutorial that is pretty good https://go.dev/doc/tutorial/generics.

You probably figured it out from the Tour of Go, but the E was just a placeholder for the type name. In the blog I was looking at they used 'E'. In the Tour of Go they use 'T', but it could be anything.